testing/driver/driver-testing/src/integrationTest/kotlin/net/corda/testing/driver/tests/ConsensualLedgerTests.kt

```diff
@@ -0,0 +1,143 @@
+import com.fasterxml.jackson.databind.SerializerProvider
+import com.fasterxml.jackson.databind.module.SimpleModule
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.r3.corda.demo.consensual.ConsensualDemoFlow
+import com.r3.corda.demo.consensual.FindTransactionFlow
+import com.r3.corda.demo.consensual.FindTransactionParameters
+import com.r3.corda.demo.consensual.FindTransactionResponse
+import com.r3.corda.demo.consensual.TestConsensualStateResult
+import java.util.concurrent.TimeUnit.MINUTES
+import net.corda.crypto.core.parseSecureHash
+import net.corda.testing.driver.AllTestsDriver
+import net.corda.v5.base.types.MemberX500Name
+import net.corda.v5.crypto.SecureHash
+import net.corda.virtualnode.VirtualNodeInfo
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.fail
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS
+import org.junit.jupiter.api.Timeout
+import org.junit.jupiter.api.extension.RegisterExtension
+import org.slf4j.LoggerFactory
+
+@Suppress("FunctionName")
+@Timeout(5, unit = MINUTES)
+@TestInstance(PER_CLASS)
+class ConsensualLedgerTests {
+    private companion object {
+        private const val TEST_INPUT = "test data"
+        private const val FAIL_INPUT = "fail"
+    }
+
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    private val alice = MemberX500Name.parse("CN=Alice, OU=Testing, O=R3, L=London, C=GB")
+    private val bob = MemberX500Name.parse("CN=Bob, OU=Testing, O=R3, L=San Francisco, C=US")
+    private val charlie = MemberX500Name.parse("CN=Charlie, OU=Testing, O=R3, L=Paris, C=FR")
+    private val consensualLedger = mutableMapOf<MemberX500Name, VirtualNodeInfo>()
+    private val jsonMapper = ObjectMapper().apply {
+        registerModule(KotlinModule.Builder().build())
+
+        val module = SimpleModule().apply {
+            addSerializer(SecureHash::class.java, SecureHashSerializer)
+            addDeserializer(SecureHash::class.java, SecureHashDeserializer)
+        }
+        registerModule(module)
+    }
+
+    @RegisterExtension
+    private val driver = AllTestsDriver(alice, bob, charlie)
+
+    @BeforeAll
+    fun start() {
+        // Ensure that we use the corda-driver bundle rather than a directory of its classes.
+        assertThat(AllTestsDriver::class.java.protectionDomain.codeSource.location.path).endsWith(".jar")
+
+        val virtualNodes = mutableSetOf<VirtualNodeInfo>()
+        driver.run { dsl ->
+            virtualNodes += dsl.startNode(setOf(alice, bob, charlie)).onEach { vNode ->
+                logger.info("VirtualNode({}): {}", vNode.holdingIdentity.x500Name, vNode)
+            }
+        }
+        logger.info("{}, {} and {} started successfully", alice.commonName, bob.commonName, charlie.commonName)
```

Comment:
can switch everywhere to string interpolation ... please 🙏
```suggestion
        logger.info("${alice.commonName}, ${bob.commonName} and ${charlie.commonName} started successfully")
```

Reply:
No thanks.

Reply:
But please explain why not ?
What's the downside ?

Reply:
I've had this conversation before with Chris before and I don't have the link he shared to hand, but https://stackoverflow.com/questions/10555409/logger-slf4j-advantages-of-formatting-with-instead-of-string-concatenation explains the issue. We do care about performance of Corda and @yarivus's version will use more resources when info level logging is disabled than what Chris wrote. That's why much of the Corda code uses the function argument (lambda) versions of logging methods, where the function is not called (I hope) and the string not constructed if the log level is disabled. The objection to that is it's not standard in the Java community, whereas what Chris wrote is probably the most common form. 

I personally feel the objection to the lambda form  not being standard and portable to Java is a minor concern given we chose Kotlin at all, so we're already moving away from the most standard choice, i.e. probably still plain Java 8 despite all the verbosity that would by Kotlin.

The interpolated form is more readable once you get used to it, which matters to me, but not enough to ignore the performance aspects.

Reply:
Thanks for the detailed explanation @dickon 👍 
BTW - this is pretty unfortunate ... other languages have optimizations around this